src/sentry/dynamic_sampling/tasks/collect_orgs.py

```diff
@@ -0,0 +1,42 @@
+from sentry_sdk import capture_message, set_extra
+
+from sentry import options
+from sentry.dynamic_sampling.tasks.common import GetActiveOrgs, TimedIterator, TimeoutException
+from sentry.dynamic_sampling.tasks.constants import MAX_PROJECTS_PER_QUERY, MAX_SECONDS
+from sentry.dynamic_sampling.tasks.logging import log_task_execution, log_task_timeout
+from sentry.dynamic_sampling.tasks.task_context import TaskContext
+from sentry.dynamic_sampling.tasks.utils import dynamic_sampling_task
+from sentry.tasks.base import instrumented_task
+
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.tasks.collect_orgs",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+    soft_time_limit=2 * 60 * 60,
+    time_limit=2 * 60 * 60 + 5,
+)
+@dynamic_sampling_task
+def collect_orgs() -> None:
+
+    enabled = options.get("dynamic_sampling.tasks.collect_orgs") or False
+
+    if not enabled:
+        return
+
+    context = TaskContext("sentry.dynamic_sampling.tasks.collect_orgs", MAX_SECONDS)
```

Comment:
I would prefer to have this configurable, so that we can tweak it in case our upper bound estimate is wrong.

Reply:
I try to keep it as light as possible, don't want to add new options that we need to delete after.
I was thinking of putting 10 min as an upper bound but then I thought that based on our experiments 1 min should handle everything.